meshtastic/interfaces/ble/receive_service.py

```
@@ -33,82 +33,151 @@
+            )
+            has_ever_connected_session_fn = self._as_usable_callable(
+                has_ever_connected_session
+            )
+            if has_ever_connected_session_fn is not None:
+                return bool(has_ever_connected_session_fn())
+        raw_ever_connected = getattr(self._iface, "_ever_connected", False)
+        if _is_unconfigured_mock_member(raw_ever_connected):
+            return False
+        return raw_ever_connected is True
+
+    def _is_connection_closing(self) -> bool:
```

Ruff (0.15.6)

[warning] Too many branches (24 > 12)

(PLR0912)